src/functions/module_logical.ts

```diff
@@ -49,27 +56,33 @@ export const IF = {
   description: _t("Returns value depending on logical expression."),
   args: [
     arg(
-      "logical_expression (boolean)",
+      "logical_expression (boolean, range<boolean>)",
       _t(
         "An expression or reference to a cell containing an expression that represents some logical value, i.e. TRUE or FALSE."
       )
     ),
-    arg("value_if_true (any)", _t("The value the function returns if logical_expression is TRUE.")),
     arg(
-      "value_if_false (any, default=FALSE)",
+      "value_if_true (any, range)",
+      _t("The value the function returns if logical_expression is TRUE.")
+    ),
+    arg(
+      "value_if_false (any, range, default=FALSE)",
       _t("The value the function returns if logical_expression is FALSE.")
     ),
   ],
-  compute: function (
-    logicalExpression: Maybe<FunctionResultObject>,
-    valueIfTrue: Maybe<FunctionResultObject>,
-    valueIfFalse: Maybe<FunctionResultObject>
-  ): FunctionResultObject {
-    const result = toBoolean(logicalExpression?.value) ? valueIfTrue : valueIfFalse;
+  compute: function (logicalExpression: Arg, valueIfTrue: Arg, valueIfFalse: Arg) {
+    if (isMultipleElementMatrix(logicalExpression)) {
+      return applyVectorization(IF.compute, [logicalExpression, valueIfTrue, valueIfFalse]);
+    }
+    let result = toBoolean(toScalar(logicalExpression)) ? valueIfTrue : valueIfFalse;
+    // useful for interpreting empty cell references as empty strings. But must be removed to make empty cell references equal to zero
```

Comment:
Could you add a dedicated test to this case so we can state clearly "we have decided to return empty stringsin this specific case" ?